src/main/scala/org/neo4j/spark/rdd/Neo4jTupleRDD.scala

```diff
@@ -14,21 +15,19 @@ class Neo4jTupleRDD(@transient sc: SparkContext, val query: String, val paramete
 
   override def compute(split: Partition, context: TaskContext): Iterator[Seq[(String, AnyRef)]] = {
     val driver: Driver = config.driver()
-    val session = driver.session()
+    val session = driver.session(config.sessionConfig())
     try {
-      val result = session.run(query, parameters.toMap.asJava)
-
-      result.asScala.map((record) => {
-        val res = record.asMap().asScala.toSeq
-        if (!result.hasNext) {
-          if (session.isOpen) session.close()
-          driver.close()
+      val txWork = new TransactionWork[Iterator[Seq[(String, AnyRef)]]] {
+        override def execute(tx: Transaction): Iterator[Seq[(String, AnyRef)]] = {
+          val result: Result = tx.run(query, parameters.toMap.asJava)
+          result.list().asScala.map((record) => {
```

Comment:
same as in the other place, i'm a bit afraid of materializing all results eagerly and not as spark consumes the iterator ...

Reply:
ping @conker84 ?